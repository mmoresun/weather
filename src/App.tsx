
import { useEffect, useState } from 'react';
import './App.css';
import SearchPanel from './components/SearchPanel/SearchPanel';
import Header from './components/Header/Header';
import WeatherCardsList from './components/WeatherCardsList/WeatherCardsList';
import { IWeatherDataObj, MyFormSubmitEvent } from './types/types';
import axios from 'axios';

const API_WEATHER_KEY = process.env.REACT_APP_WEATHER_API_KEY;

const App = () => {

  // 'Add to my list button' enabled/disabled switcher
  const [addDisabled, setAddDisabled] = useState<boolean>(false);

  // get and set value from search field
  const [myCity, setMyCity] = useState<string>('');

  const getCity = (e: MyFormSubmitEvent): void => {
    e.preventDefault();
    setMyCity(e.currentTarget.city.value);
  };

  const [weatherData, setWeatherData] = useState({
  } as IWeatherDataObj);

  const getWeather = async (myCity: string) => {
    const cityUrl = `https://api.openweathermap.org/data/2.5/weather?q=${myCity}&lang=en&appid=${API_WEATHER_KEY}&units=metric`;
    await axios
      .get(cityUrl)
      .then((response) => setWeatherData({
        name: response.data.name,
        country: response.data.sys.country,
        wind: response.data.wind.speed,
        clouds: response.data.weather[0].description,
        temp: Math.round(response.data.main.temp),
        feels_like: Math.round(response.data.main.feels_like),
        pressure: response.data.main.pressure,
        humidity: response.data.main.humidity,
        icon: response.data.weather[0].icon,
        error: undefined
      }))
      .catch(() => setWeatherData({
        name: undefined,
        country: undefined,
        wind: undefined,
        clouds: undefined,
        temp: undefined,
        feels_like: undefined,
        pressure: undefined,
        humidity: undefined,
        icon: undefined,
        error: `City "${myCity}" is not found`
      }));

    setAddDisabled(false);
  };

  useEffect(() => {
    // if something entered to the search string, run getWeather function
    myCity && getWeather(myCity);

  }, [myCity]);

  // the first render
  useEffect(() => {

    getWeather('Kyiv');

  }, []);

  return (
    <div className="App">
      <Header />
      <SearchPanel
        getCity={getCity}
        setMyCity={setMyCity}
      />
      <WeatherCardsList
        addDisabled={addDisabled}
        setAddDisabled={setAddDisabled}
        weatherData={weatherData}
      />
    </div>
  );
}

export default App;
