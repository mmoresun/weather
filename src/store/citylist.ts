
// initial state

import { v4 as uuid } from 'uuid';
import { ICityStateEntry } from '../types/types';

export const citylist: ICityStateEntry[] = [
  {
    url: 'https://api.openweathermap.org/data/2.5/weather?q=Kyiv&lang=en&appid=08f2a575dda978b9c539199e54df03b0&units=metric',
    id: uuid()
  },
  {
    url: 'https://api.openweathermap.org/data/2.5/weather?q=Warsaw&lang=en&appid=08f2a575dda978b9c539199e54df03b0&units=metric',
    id: uuid()
  },
  {
    url: 'https://api.openweathermap.org/data/2.5/weather?q=Prague&lang=en&appid=08f2a575dda978b9c539199e54df03b0&units=metric',
    id: uuid()
  },
  {
    url: 'https://api.openweathermap.org/data/2.5/weather?q=Antalya&lang=en&appid=08f2a575dda978b9c539199e54df03b0&units=metric',
    id: uuid()
  },
  {
    url: 'https://api.openweathermap.org/data/2.5/weather?q=Paris&lang=en&appid=08f2a575dda978b9c539199e54df03b0&units=metric',
    id: uuid()
  }
]
