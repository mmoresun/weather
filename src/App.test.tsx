
import renderer from 'react-test-renderer';
import App from "./App";
import { Provider } from 'react-redux';
import { store } from './store/store';
import { render, screen, waitFor } from '@testing-library/react';

it('useEffect works, getWeather calls', async () => {
  render(<Provider store={store}><App /></Provider>);
  await waitFor(() => {
    expect(screen.getByText(/City or region/i)).toBeInTheDocument()    
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><App /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// ========================

// INITIAL TEST
// import React from 'react';
// import { render, screen } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

