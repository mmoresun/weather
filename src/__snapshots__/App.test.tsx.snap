// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="MuiBox-root css-1qk4bij"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-py75jh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1xouml8-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
        <div
          className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          <a
            href="https://mmoresun.github.io/weather"
            style={
              Object {
                "color": "white",
                "textDecoration": "none",
              }
            }
          >
            â›… Weather Anywhere
          </a>
        </div>
        <a
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit css-u44ztt-MuiButtonBase-root-MuiButton-root"
          href="https://github.com/mmoresun/weather/tree/master"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          target="_blank"
        >
          GitHub
        </a>
      </div>
    </header>
  </div>
  <div
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    <form
      className="MuiBox-root css-0"
      name="searchpanel"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            name="city"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter city"
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Get weather
      </button>
    </form>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <p
      style={
        Object {
          "display": "none",
        }
      }
    >
      <b />
    </p>
    <div
      className="MuiBox-root css-rwf8oj"
    >
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-193eagd-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Weather at 
            <b />
            :
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            ðŸŒ¡ Temperature: 
            <b>
              Â°C
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            â˜´ Wind: 
            <b>
               m/s
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Sky: 
            <b />
             
          </p>
          <img
            alt=""
            src="https://openweathermap.org/img/wn/undefined@2x.png"
          />
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Details
          </button>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-10r5to9-MuiCardActions-root"
        >
          <div
            className="MuiBox-root css-1py8c8m"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Refresh â†»
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-10fi992-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-193eagd-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Weather at 
            <b />
            :
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            ðŸŒ¡ Temperature: 
            <b>
              Â°C
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            â˜´ Wind: 
            <b>
               m/s
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Sky: 
            <b />
             
          </p>
          <img
            alt=""
            src="https://openweathermap.org/img/wn/undefined@2x.png"
          />
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Details
          </button>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-10r5to9-MuiCardActions-root"
        >
          <div
            className="MuiBox-root css-1py8c8m"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Refresh â†»
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-10fi992-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-193eagd-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Weather at 
            <b />
            :
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            ðŸŒ¡ Temperature: 
            <b>
              Â°C
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            â˜´ Wind: 
            <b>
               m/s
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Sky: 
            <b />
             
          </p>
          <img
            alt=""
            src="https://openweathermap.org/img/wn/undefined@2x.png"
          />
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Details
          </button>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-10r5to9-MuiCardActions-root"
        >
          <div
            className="MuiBox-root css-1py8c8m"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Refresh â†»
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-10fi992-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-193eagd-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Weather at 
            <b />
            :
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            ðŸŒ¡ Temperature: 
            <b>
              Â°C
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            â˜´ Wind: 
            <b>
               m/s
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Sky: 
            <b />
             
          </p>
          <img
            alt=""
            src="https://openweathermap.org/img/wn/undefined@2x.png"
          />
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Details
          </button>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-10r5to9-MuiCardActions-root"
        >
          <div
            className="MuiBox-root css-1py8c8m"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Refresh â†»
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-10fi992-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-193eagd-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Weather at 
            <b />
            :
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            ðŸŒ¡ Temperature: 
            <b>
              Â°C
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            â˜´ Wind: 
            <b>
               m/s
            </b>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Sky: 
            <b />
             
          </p>
          <img
            alt=""
            src="https://openweathermap.org/img/wn/undefined@2x.png"
          />
        </div>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Details
          </button>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-10r5to9-MuiCardActions-root"
        >
          <div
            className="MuiBox-root css-1py8c8m"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Refresh â†»
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-10fi992-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
